test/src/org/labkey/test/components/cds/ChangePasswordDialog.java

```diff
@@ -0,0 +1,79 @@
+package org.labkey.test.components.cds;
+
+import org.labkey.test.BaseWebDriverTest;
+import org.labkey.test.Locator;
+import org.labkey.test.WebDriverWrapper;
+import org.labkey.test.components.Component;
+import org.labkey.test.components.WebDriverComponent;
+import org.labkey.test.components.html.Input;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+public class ChangePasswordDialog extends WebDriverComponent<Component.ElementCache>
+{
+    private final WebElement _dialogEl;
+    private final WebDriverWrapper _webDriverWrapper;
+
+    public ChangePasswordDialog(BaseWebDriverTest test)
+    {
+        _webDriverWrapper = test;
+        _dialogEl = Locator.tagWithAttribute("div", "data-form", "account-change-password").refindWhenNeeded(_webDriverWrapper.getDriver());
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return _dialogEl;
+    }
+
+    @Override
+    protected WebDriver getDriver()
+    {
+        return _webDriverWrapper.getDriver();
+    }
+
+    public ChangePasswordDialog setPreviousPassword(String value)
+    {
+        newElementCache().previousPwd.set(value);
+        return this;
+    }
+
+    public ChangePasswordDialog setPassword(String value)
+    {
+        newElementCache().password.set(value);
+        return this;
+    }
+
+    public ChangePasswordDialog setReEnterPassword(String value)
+    {
+        newElementCache().reEnterPassword.set(value);
+        return this;
+    }
+
+    public String getErrorMessage()
+    {
+        return newElementCache().notification.getText();
+    }
+
+    public void submit()
+    {
+        newElementCache().submitButton.click();
+    }
```

Comment:
`submit` should wait for the dialog to disappear or go stale. There should be a separate `submitExpectingError` method for the error test cases to use.